### PR TITLE
Applied style guide to test names

### PR DESCRIPTION
**Pull request by doublemarked**
@superkhau @Amir-61 @bajtos,

Following from our discussion in PR #1142 - as an exercise, I went through all the tests in loopback-dl.test.js and changed the names to match the style guide. I did add nested describe() blocks in a couple places but generally left the organization of tests untouched. Probably some groups of tests should be grouped better.

This involved a number of small decisions on terminology which may or may not meet your expectations. I do kind of wonder whether the process of you guys reviewing and approving a PR like this will exceed the time it took me to perform it (just under an hour). Let's see how you guys feel about it.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: test/loopback-dl.test.js [hunk 1]
```diff
@@ -661,8 +661,8 @@ describe('DataSource define model', function() {
 
 });
 
-describe('Load models with base', function() {
-  it('should set up base class via base option', function() {
+describe('Models with Base Model', function() {
+  it('have a base class according to the base option', function() {
```
Comment (bajtos) [inline test/loopback-dl.test.js]: `it('has...`

File: test/loopback-dl.test.js [hunk 2]
```diff
@@ -698,7 +698,7 @@ describe('Load models with base', function() {
     Customer.definition.properties.name.should.have.property('type', String);
   });
 
-  it('should inherit properties by clone from base option', function() {
+  it('inherit properties by clone from base model', function() {
```
Comment (bajtos) [inline test/loopback-dl.test.js]: `it('inherits...`

There are few more places below that don't use third-person singular and should be fixed too.

File: test/loopback-dl.test.js [hunk 3]
```diff
@@ -775,174 +775,177 @@ describe('Models attached to a dataSource', function() {
...
+    it('updates instances', function(done) {
+      Post.create({title: 'a', content: 'AAA'}, function(err, post) {
+        post.title = 'b';
+        Post.updateOrCreate(post, function(err, p) {
+          should.not.exist(err);
           p.id.should.be.equal(post.id);
-          should.not.exist(p._id);
           p.content.should.be.equal(post.content);
-          p.title.should.be.equal('b');
+          should.not.exist(p._id);
```
Comment (bajtos) [inline test/loopback-dl.test.js]: Telling `git` to ignore white space changes removes this artefact: https://github.com/strongloop/loopback-datasource-juggler/pull/1150/files?w=1

Unfortunately github does not allow comments in `w=1` mode :(

**Comment by bajtos** [comment]
@slnode ok to test

File: test/loopback-dl.test.js (hunk 1 above)
Comment (doublemarked) [inline test/loopback-dl.test.js]: This is due to plurality in the describe, `Models`. I'll reword the describe block to singular.

File: test/loopback-dl.test.js (hunk 2 above)
Comment (doublemarked) [inline test/loopback-dl.test.js]: Thanks.

**Comment by Amir-61** [comment]
>I'l leave it to functional-area-owners ( @superkhau, @Amir-61, anybody else?) to review the patch in more details.

I'll take care of the review from here. Thanks @bajtos

File: test/loopback-dl.test.js [hunk 4]
```diff
@@ -1560,35 +1563,35 @@ describe('DataAccessObject', function() {
...
 
-  it('should skip conversion if a simple property produces NaN for numbers',
+  it('does not coerce to a number a simple value that produces NaN',
     function() {
       where = model._coerce({age: 'xyz'});
       assert.deepEqual(where, {age: 'xyz'});
     });
 
-  it('should skip conversion if an array property produces NaN for numbers',
+  it('does not coerce to a number a simple value in an array that produces NaN',
```
Comment (Amir-61) [inline test/loopback-dl.test.js]: `does not coerce to a number for a simple value in an array that produces NaN`

File: test/loopback-dl.test.js [hunk 5]
```diff
@@ -1658,7 +1661,7 @@ describe('Load models from json', function() {
     assert.equal(models.Address.dataSource, ds);
   });
 
-  it('should allow customization of default model base class', function() {
+  it('applies a default model base class', function() {
```
Comment (Amir-61) [inline test/loopback-dl.test.js]: I would say: `allow customization of default model base class` is more clear

**Comment by Amir-61** [comment]
@doublemarked Reviewed the patch... I left a few minor suggestions... Please apply them and @bajtos's suggestions and rebase your branch on top of master.

Thanks for contribution!

File: test/loopback-dl.test.js [hunk 6]
```diff
@@ -775,174 +775,177 @@ describe('Models attached to a dataSource', function() {
...
+          Post.findById(post.id, function(err, p) {
+            p.id.should.be.equal(post.id);
+            should.not.exist(p._id);
+            p.content.should.be.equal(post.content);
+            p.title.should.be.equal('b');
+
+            done();
+          });
         });
+
```
Comment (superkhau) [inline test/loopback-dl.test.js]: remove empty line

File: test/loopback-dl.test.js [hunk 7]
```diff
@@ -775,174 +775,177 @@ describe('Models attached to a dataSource', function() {
...
+            p.id.should.be.equal(post.id);
+            should.not.exist(p._id);
+            p.content.should.be.equal(post.content);
+            p.title.should.be.equal('a');
+            p.comments.length.should.be.equal(1);
+            p.comments[0].should.be.equal('Comment1');
+            done();
+          });
+        });
+
```
Comment (superkhau) [inline test/loopback-dl.test.js]: remove empty line

File: test/loopback-dl.test.js [hunk 8]
```diff
@@ -775,174 +775,177 @@ describe('Models attached to a dataSource', function() {
...
+        Post.findById(p.id, function(err, p) {
           p.id.should.be.equal(post.id);
           should.not.exist(p._id);
           p.content.should.be.equal(post.content);
-          p.title.should.be.equal('a');
-          p.comments.length.should.be.equal(1);
-          p.comments[0].should.be.equal('Comment1');
+          p.title.should.be.equal(post.title);
+          p.id.should.be.equal(post.id);
+
```
Comment (superkhau) [inline test/loopback-dl.test.js]: remove empty line

File: test/loopback-dl.test.js [hunk 9]
```diff
@@ -775,174 +775,177 @@ describe('Models attached to a dataSource', function() {
...
-        should.not.exist(p._id);
-        p.content.should.be.equal(post.content);
-        p.title.should.be.equal(post.title);
-        p.id.should.be.equal(post.id);
+          Post.findById(post.id, function(err, p) {
+            p.id.should.be.equal(post.id);
+            should.not.exist(p._id);
+            p.content.should.be.equal(post.content);
+            p.title.should.be.equal('b');
+
```
Comment (superkhau) [inline test/loopback-dl.test.js]: remove empty line

File: test/loopback-dl.test.js [hunk 10]
```diff
@@ -775,174 +775,177 @@ describe('Models attached to a dataSource', function() {
...
-        done();
-      });
     });
-
   });
 });
 
 describe('DataSource connector types', function() {
-  it('should return an array of types', function() {
+  it('return an array of types using getTypes', function() {
```
Comment (superkhau) [inline test/loopback-dl.test.js]: returns

File: test/loopback-dl.test.js [hunk 11]
```diff
@@ -1172,7 +1175,7 @@ describe('Load models with relations', function() {
     done();
   });
 
-  it('should throw if a relation is missing type', function(done) {
+  it('throw an error if a relation is missing type', function(done) {
```
Comment (superkhau) [inline test/loopback-dl.test.js]: throws

File: test/loopback-dl.test.js [hunk 12]
```diff
@@ -1187,7 +1190,7 @@ describe('Load models with relations', function() {
 
   });
 
-  it('should throw if the relation type is invalid', function(done) {
+  it('throw an error if a relation type is invalid', function(done) {
```
Comment (superkhau) [inline test/loopback-dl.test.js]: throws

File: test/loopback-dl.test.js [hunk 13]
```diff
@@ -1273,7 +1276,7 @@ describe('Load models with relations', function() {
     done();
   });
 
-  it('should set up relations after attach', function(done) {
+  it('set up relations after attach', function(done) {
```
Comment (superkhau) [inline test/loopback-dl.test.js]: sets

File: test/loopback-dl.test.js [hunk 14]
```diff
@@ -1560,35 +1563,35 @@ describe('DataAccessObject', function() {
...
   });
 
-  it('should skip undefined values', function() {
+  it('does not coerce undefined values', function() {
     where = model._coerce({date: undefined});
     assert.deepEqual(where, {date: undefined});
   });
 
-  it('should skip conversion if a simple property produces NaN for numbers',
+  it('does not coerce to a number a simple value that produces NaN',
```
Comment (superkhau) [inline test/loopback-dl.test.js]: does not coerce value to a Number given a simple value that produces NaN

File: test/loopback-dl.test.js (hunk 4 above)
Comment (superkhau) [inline test/loopback-dl.test.js]: does not coerce value to a number for a simple value in an array that produces NaN

File: test/loopback-dl.test.js [hunk 15]
```diff
@@ -1691,7 +1694,7 @@ describe('Load models from json', function() {
     assert(Customer.prototype instanceof User);
   });
 
-  it('should be able to extend models', function(done) {
+  it('extends models', function(done) {
```
Comment (superkhau) [inline test/loopback-dl.test.js]: allows model extension

File: test/loopback-dl.test.js [hunk 16]
```diff
@@ -1735,7 +1738,7 @@ describe('Load models from json', function() {
     done(null, customer);
   });
 
-  it('should be able to extend models with merged settings', function(done) {
+  it('extend models with merged settings', function(done) {
```
Comment (superkhau) [inline test/loopback-dl.test.js]: allows model extension with merged settings

**Comment by superkhau** [comment]
@doublemarked Thanks for taking this on and updating a lot of text to prevent future confusion. This is good enough for now (there are some other vocab and refactorings needed, but this is good enough for now). @Amir-61 You can take it from here and land -- no more review needed from me.

**Comment by doublemarked** [comment]
@Amir-61 all feedback applied. thanks.

**Comment by Amir-61** [comment]
There was a commit message issue which I fixed and landed with https://github.com/strongloop/loopback-datasource-juggler/commit/a65311dc7cc2a011d67be479bf20e368ae90981e ; we do not use past tense like `Applied style guide to test names` for commit messages. Also, added more info in the commit message. Thanks for the patch.

**Comment by doublemarked** [comment]
tyvm @Amir-61

**Comment by Amir-61** [comment]
@doublemarked Thank you for the contribution!

**Pull request closed without merging**